src/Raven.Client/Documents/Session/Operations/QueryOperation.cs

```diff
@@ -165,34 +165,39 @@ public void EnsureIsAcceptableAndSaveResult(QueryResult result)
             if (result == null)
                 throw new IndexDoesNotExistException("Could not find index " + _indexName);
 
-            if (_indexQuery.WaitForNonStaleResults && result.IsStale)
+            EnsureIsAcceptable(result, _indexQuery.WaitForNonStaleResults, _sp, _session);
+
+            _currentQueryResults = result;
+            //_currentQueryResults.EnsureSnapshot();
```

Comment:
why is this commented?

Reply:
no longer valid code - removed in https://github.com/ravendb/ravendb/pull/4681